control/metrics.go

```diff
@@ -453,6 +453,14 @@ func (mc *metricCatalog) RmUnloadedPluginMetrics(lp *loadedPlugin) {
 	mc.mutex.Lock()
 	defer mc.mutex.Unlock()
 	mc.tree.DeleteByPlugin(lp)
+
+	// Update metric catalog keys
+	mc.keys = []string{}
+	mts := mc.tree.gatherMetricTypes()
+	for _, m := range mts {
+		mc.keys = append(mc.keys, m.Namespace().Key())
+	}
```

Comment:
I definitely think they should be killed if unloaded. In case of a security risk or bug an unload is a safety valve for "do not use".